### PR TITLE
Fix mix node

### PR DESCRIPTION
**Pull request by AnthonyLaw**
The node info from `/node/peers` can't be trusted, it always gives the wrong information, causing the node list to mix with another network.

### Add
- new `networkGenerationHashSeed` property in NodeMonitor class
- add `networkGenerationHashSeed` on add node filter, to ensure always get the right node.

### Removed
- delete env `NETWORK_IDENTIFIER` because network type is always collected from the provided node.

### Update
- rename `fetchNodesByURL` -> `fetchNodePeersByURL`
- `fetchNodePeersByURL` method will only collect nodes from the node/peers.
- refactor on `getNodeListInfo` method, collect nodes info (only API roles) without condition.

### CODE REVIEW

File: src/services/NodeMonitor.ts [hunk 1]
```diff
@@ -54,10 +57,10 @@ export class NodeMonitor {
 			this.isRunning = true;
 			this.clear();
 
-			await this.getNetworkType(); // Read network Type from provided nodes.
+			await this.getNetworkInfo(); // Read network Type and generated hash seed from provided nodes.
```
Comment (yilmazbahadir) [inline src/services/NodeMonitor.ts:60]: Should we rename this method as it's not a getter method (doesn't return anything)?
How about `fetchAndSetNetworkInfo`?

File: src/services/NodeMonitor.ts [hunk 2]
```diff
@@ -78,19 +81,21 @@ export class NodeMonitor {
...
 			this.addNodesToList(peers);
 		}
 
 		// Nested fetch node list from current nodeList[]
 		const nodeListPromises = this.nodeList.map(async (node) => {
 			if (isAPIRole(node.roles)) {
-				return this.fetchNodesByURL(getNodeURL(node, monitor.API_NODE_PORT));
+				const hostUrl = await ApiNodeService.buildHostUrl(node.host);
+
+				return this.fetchNodePeersByURL(hostUrl);
```
Comment (Wayonb) [inline src/services/NodeMonitor.ts:98]: Maybe I miss it, are pulling ``node/peers`` from every api node?

**Review by Wayonb — CHANGES_REQUESTED** [review]

File: src/services/NodeMonitor.ts (hunk 2 above)
Comment (AnthonyLaw) [reply src/services/NodeMonitor.ts:98]: Yes, every API node.

**Review by Wayonb — APPROVED** [review]
Good job

**Review by yilmazbahadir — APPROVED** [review]

**Pull request merged**